### DISCUSSION TITLE
It is impossible to reconnect a device to tuya-local if the Wi-Fi network on the device has changed

### DISCUSSION THREAD

**Discussion by garry0garry** [make-all/tuya-local#1112] [General]
https://github.com/make-all/tuya-local/issues/1111

It is impossible to reconnect a device to [tuya-local](https://github.com/make-all/tuya-local) if the Wi-Fi network on the device has changed. I solved the problem via the terminal. Replaced the old "device_id" and "local_key" to new ones in the files:
.storage/core.entity_registry
.storage/core.device_registry

Why can't this problem be solved through the GUI?

**Comment by make-all** [comment]
The device id should not be changing.  The device id is used as the unique id for the device, and is scattered throughout the HA config once the device is added, so it cannot simply be updated.  If your device id is really changing, it is a different device, and the device needs to be removed and re-added.
Only the local key and IP address can be updated through the UI.  Possibly bug #1110 is affecting the saving of these in the current release (and previous releases since sub-device support was added).

**Comment by bilogic** [comment]
I'm facing the same issue, how do I update the IP address?

**Reply by bilogic** [reply]
![image](https://github.com/make-all/tuya-local/assets/946010/afffe260-1a82-4d44-9962-45bff8e0e1c6)

My router issued different addresses to my devices after it rebooted.
I read the previous IP addresses from `.storage/core.config_entires` and configured the router to forever maintain them.

But HA still says `Not loaded`, what can I do?

**Reply by bilogic** [reply]
After several cycles of updating tuya-local and updating/rebooting HA, all my devices are back online.

**Comment by garry0garry** [comment]
The ip address has not changed.  When changing the access point, the id and key changed.

**Reply by bilogic** [reply]
I'm thinking, if the devices are defining their key and ID based on the previous router, most likely it is the router's MAC address of the SSID they are connected to. Why not try spoofing your previous MAC address and make the SSID exactly the same?

**Comment by garry0garry** [comment]
I did not change the router (on which DHCP is running), I changed the Wi-Fi access point on the device within the same local network. MAC did not change, IP did not change, Device ID and Key changed.

**Reply by bilogic** [reply]
Changing the AP will lead to the tuya device seeing a different MAC. So I would try modifying the new AP MAC to be same as the old. If you did, then I have no other ideas what might have caused the tuya ID to change.

**Comment by garry0garry** [comment]
> The device id is used as the unique id for the device

TOMN TOMPD-63LW

Before reset:
`"id": "bfc1a579afecf788bcfkqs"`

After reset:
`"id": "bf6882f2de238b3d4bqboa"`

**Comment by make-all** [comment]
It seems some devices will change their device id when reset.

https://support.tuya.com/en/help/_detail/K9hrdz9omvknb

This makes them a different device as far as tuya-local is concerned, so you need to remove the old device and set up a new one.

**Reply by bilogic** [reply]
Thanks, btw is there a way to change IP address without removing and adding? I would the previously collected data to persist.

**Reply by make-all** [reply]
Yes, going into the Integrations Settings and clicking "Tuya local" will take you to a list of Tuya local devices with [CONFIGURE] buttons next to each one.  The  IP address and local key can be changed there, just not the device id, since that is a unique id for the device.

**Reply by bilogic** [reply]
ok thanks, for some reason, my list of Tuya devices reported as `Not loaded` (See my screenshot). At that point in time, the IP address could not be shown nor edited. That left me in quite a bind. I ended up trying many things and don't really know what I did to solve it.

**Comment by garry0garry** [comment]
In this case, I will have to change the automation settings. I replaced 2 values in the files:
.storage/core.entity_registry
.storage/core.device_registry
I don't see any problems. What is the risk of adding an optional ID change field in the integration?
